pages/docs/reference/compatibility.md

```diff
@@ -0,0 +1,114 @@
+
+## Compatibility across platforms
+ 
+Kotlin is available for several platforms (JVM/Android, JavaScript and the upcoming native platforms). Every platform has its own peculiarities (e.g. JavaScript has no proper integers), so we have to adapt the language accordingly. Our goal is not to provide reasonable code portability without sacrificing too much.
+  
+Every platform may feature specific language extensions (such as platform types for JVM and dynamic types for JavaScript) or restrictions (e.g. some overloading-related restrictions on  the JVM), but the core language remains the same. 
+
+The Standard Library provides a core API available on all platforms, and we strive to make these APIs work in the same way on every platform. Along with these, the Standard Library provides platform-specific extensions (e.g. `java.io` for JVM, or `js()` for JavaScript), plus some APIs that can be called uniformly, but work differently (such as regular expressions for JVM and JavaScript).    
+
+## Experimental features
+
+Experimental features, such as coroutines in Kotlin 1.1, have exemption from the compatibility modes listed above. Such features require an opt-in to use without compiler warning. Experimental features are at least backwards compatible for patch version updates, but we do guarantee no compatibility for minor version updates (migration aids will be provided where possible).    
+
+| EXP  | 1.1 | 1.1.X | 1.2 | 1.2.X | 
+|---:|:---:|:---:|:---:|:---:|
+| **1.1** | - | BC | NO | NO  
+| **1.1.X** | BC | - | NO | NO
+| **1.2** | NO | NO | - | BC 
+| **1.2.X** | NO | NO | BC | - 
+
+## EAP builds
+
+We publish Early Access Preview (EAP) builds to special channels where early adopters from the community can try them out and give us feedback. Such builds provide no compatibility guarantees whatsoever (although we do our best to keep them reasonably compatible with releases and with each other). Quality expectations for such builds are also much lower than for releases. Beta builds also fall under this category.
+
+**IMPORTANT NOTE**: all binaries compiled by EAP builds for BCB-version (e.g. 1.1-eap-X) are **rejected by release builds of the compiler**. We don't want any code compiled by pre-release versions to be kept around after a stable version is released. 
+
+## Compatibility modes
+
+When a big team is migrating onto a new version, it may appear in a "inconsistent state" at some point, when some developers have already updated, and others haven't. To prevent the former from writing and committing code that others may not be able to compile, we provide the following command line switches (also available in the IDE and build systems):   
+
+- `-language-version`=X.Y - compatibility mode for Kotlin language version X.Y, reports errors for all language features that came out later
+- `-api-version`=X.Y - compatibility mode for Kotlin API version X.Y, reports errors for all code using newer APIs (including the code generated by the compiler).
```

Comment:
> code using newer APIs

"code using declarations from newer standard Kotlin libraries" or something like this? To highlight that this only applies to our libraries, not other libraries compiled with a version greater than X.Y